react/src/components/MaintenanceSettingList.tsx

```diff
@@ -0,0 +1,150 @@
+import { useSetBAINotification } from '../hooks/useBAINotification';
+import { usePainKiller } from '../hooks/usePainKiller';
+import SettingList, { SettingGroup } from './SettingList';
+import { RedoOutlined } from '@ant-design/icons';
+import { Button } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+const MaintenanceSettingList = () => {
+  const [isRecalculating, setIsRecalculating] = useState(false);
+  const [isRescanning, setIsRescanning] = useState(false);
+
+  const { t } = useTranslation();
+  const { upsertNotification } = useSetBAINotification();
+  const baiClient = useSuspendedBackendaiClient();
+  const painKiller = usePainKiller();
+
+  const recalculateUsage = () => {
+    setIsRecalculating(true);
+
+    upsertNotification({
+      message: t('maintenance.RecalculateUsage'),
+      description: t('maintenance.Recalculating'),
+      open: true,
+      backgroundTask: {
+        status: 'pending',
+        promise: baiClient.maintenance
+          .recalculate_usage()
+          .finally(() => setIsRecalculating(false)),
+        statusDescriptions: {
+          pending: t('maintenance.Recalculating'),
+          resolved: t('maintenance.RecalculationFinished'),
+          rejected: t('maintenance.RecalculationFailed'),
+        },
+      },
+    });
+  };
+
+  const rescanImages = async () => {
+    setIsRescanning(true);
+    const notiKey = upsertNotification({
+      message: t('maintenance.RescanImages'),
+      description: t('maintenance.RescanImageScanning'),
+      open: true,
+      backgroundTask: {
+        status: 'pending',
+      },
+    });
+    try {
+      const { rescan_images } = await baiClient.maintenance.rescan_images();
+      if (rescan_images.ok) {
+        upsertNotification({
+          key: notiKey,
+          backgroundTask: {
+            status: 'pending',
+            percent: 0,
+            taskId: rescan_images.task_id,
+            statusDescriptions: {
+              pending: t('maintenance.RescanImageScanning'),
+              resolved: t('maintenance.RescanImageFinished'),
+              rejected: t('maintenance.RescanFailed'),
+            },
+          },
+        });
+      } else {
+        upsertNotification({
+          key: notiKey,
+          description: t('maintenance.RescanFailed'),
+          backgroundTask: {
+            status: 'rejected',
+          },
+          duration: 1,
+        });
+      }
+    } catch (err: any) {
+      if (err && err.message) {
+        upsertNotification({
+          message: err.message,
+          description: painKiller.relieve(err.title),
+          open: true,
+        });
+      }
+      upsertNotification({
+        key: notiKey,
+        backgroundTask: {
+          status: 'rejected',
+        },
+        duration: 1,
+      });
+    } finally {
+      setIsRescanning(false);
+    }
+  };
+
+  const settingGroupList: SettingGroup = [
+    {
+      title: t('maintenance.Fix'),
```

Comment:
I'm not sure that Fix is sufficiently representative of those features (image rescan, resource recalculation). Asking @YEONFEEL96 for advice seems like a good idea.